src/model.F90

```diff
@@ -1,29 +1,80 @@
-! the host model
-program model
+module demo_rad_mod
+
+   implicit none
+   private
+
+   public :: demo_rad
+
+CONTAINS
+
+   !> \section arg_table_demo_rad  Argument Table
+   !! \htmlinclude arg_table_demo_rad.html
+   subroutine demo_rad()
+      use demo_rad_ccpp_cap, only: demo_rad_ccpp_physics_initialize
+      use demo_rad_ccpp_cap, only: demo_rad_ccpp_physics_timestep_initial
+      use demo_rad_ccpp_cap, only: demo_rad_ccpp_physics_run
+      use demo_rad_ccpp_cap, only: demo_rad_ccpp_physics_timestep_final
+      use demo_rad_ccpp_cap, only: demo_rad_ccpp_physics_finalize
 
-  use musica_aerosol,                  only : aerosol_t
-  use musica_aerosol_modal,            only : aerosol_modal_t
-  use musica_radiation,                only : radiation_initialize, radiation_run
+      integer                   :: i_time
+      integer                   :: errcode
+      integer                   :: col_start = 1
+      integer                   :: col_end = 1
+      character(len=512)        :: errmsg
 
-  implicit none
+      ! Use the suite information to setup the run
+      call demo_rad_ccpp_physics_initialize('toy_suite', errmsg, errcode)
+      if (errcode /= 0) then
+         write(6, *) trim(errmsg)
+         write(6, *) 'An error occurred in ccpp_physics_init, Exiting...'
+         stop
+      end if
 
-  class(aerosol_t), pointer :: aerosol
-  integer :: i_time
+      do i_time = 1, 5
+         ! Initialize the timestep
+         call demo_rad_ccpp_physics_timestep_initial('toy_suite',             &
+              errmsg, errcode)
+         if (errcode /= 0) then
+            write(6, *) trim(errmsg)
+            write(6, *) 'An error occurred in ccpp_physics_timestep_init, ",  &
+                 "Exiting...'
+            stop
+         end if
 
-  ! with CCPP, the aerosol_t object would probably be registered by an aerosol scheme
-  ! and exist as a standard-named variable?
-  aerosol => aerosol_modal_t()
```

Comment:
it does not appear that the modal aerosol constructor is ever called in the new version of the code

Reply:
great! yeah, I should have used that somewhere in the original code